src/actions/GraphDataActions.ts

```diff
@@ -72,6 +73,13 @@ const decorateGraphData = (graphData: any) => {
   return graphData;
 };
 
+let lastCancelableRequest = makeCancelablePromise(Promise.resolve());
+const setCurrentRequest = (promise: Promise<any>) => {
+  lastCancelableRequest.cancel();
+  lastCancelableRequest = makeCancelablePromise(promise);
+  return lastCancelableRequest.promise;
+};
```

Comment:
I think something like this is already available using the CancelablePromises utility, see https://github.com/kiali/kiali-ui/blob/2caa94b16212adc3d240ed8e160e0d1c5d895a0f/src/utils/CancelablePromises.ts#L37

Reply:
Will check that out

Reply:
Did this